src/adaptation/icap/Xaction.h

```diff
@@ -137,6 +137,7 @@ class Xaction: public Adaptation::Initiate
     bool isRetriable;  ///< can retry on persistent connection failures
     bool isRepeatable; ///< can repeat if no or unsatisfactory response
     bool ignoreLastWrite;
+    bool waitingForDns; ///< expecting a ipcache_nbgethostbyname() callback
```

Comment:
Please use C++11 initialization for new members whenever possible.

Reply:
> Please use C++11 initialization for new members whenever possible.

I do not think this PR should use default member initializer _here_. The glaring inconsistency would mislead the reader into suspecting that other class members are left uninitialized (or that their initialization depends on constructor parameters). I agree that we should use default member initializers in most new classes, but I doubt we should create such inconsistencies in old ones, especially where a single explicit class constructor makes it easy to initialize consistently.

Let's wait for [automated](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-default-member-init.html) polishing of such cases. There was already a semi-approved [RFC](http://lists.squid-cache.org/pipermail/squid-dev/2020-April/009569.html) regarding that overall approach, and this modernization should be one of the items added to the candidate list.

Reply:
I disagree with most of your points. The other members *should* be considered uninitialized until constructor initializer list runs. At least until such time as they get converted to earlier initialization.

but do not insist on this change so resolving for now.